src/main/java/base/VariableReader.java

```diff
@@ -0,0 +1,46 @@
+package base;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.BodyDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import com.github.javaparser.ast.body.VariableDeclarator;
+
+/**
+ * Class for reading variables from a given java class (CompilationUnit).
+ */
+public final class VariableReader {
```

Comment:
Is this class only used for mocking? or could it be used in other areas?

Reply:
It isn't used for mocking! It is used in the singletonverifier